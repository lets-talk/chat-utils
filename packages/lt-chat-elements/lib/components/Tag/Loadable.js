import React from 'react';
import path from 'path';
import Loadable from 'react-loadable';
import LoadingComponent from '../Loading';

const LoadableTag = Loadable({
  loader: () => import('./index'),
  loading: LoadingComponent,
  serverSideRequirePath: path.resolve(__dirname, './index'),
});

const Tag = (props) => (<LoadableTag {...props} />);

export default Tag;
