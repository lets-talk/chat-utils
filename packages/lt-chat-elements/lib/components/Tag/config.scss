@import '../../common/styles/variables';
@import '../../common/styles/animations.scss';
@import "../../common/styles/themes/themify";

$widgetBox-MessageList-background: $color-white;
