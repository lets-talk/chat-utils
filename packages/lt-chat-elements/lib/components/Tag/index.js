import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledTag = styled.span`
  color: ${(props) => props.theme.typography.sizes.caption.normal.dark.primary.color};
  font-size: ${(props) => props.theme.typography.sizes.caption.normal.dark.primary.fontSize};
  font-weight: ${(props) => props.theme.typography.sizes.caption.normal.dark.primary.fontWeight};
  background-color: ${(props) => props.theme.palette.background.secondary};

  margin: 5px 2px;
  padding: 5px;
  border-radius: 2px;
`;

class Tag extends Component {
  onClick(item, i, e) {
    if (this.props.clickItem instanceof Function) {
      this.props.clickItem(item, i, e);
    }
  }

  render() {
    const { tag } = this.props;
    return (
      <StyledTag>
        {tag.name}
      </StyledTag>
    );
  }
}

Tag.propTypes = {
  /**
   * tags is an array of tag objects
   * that will be displayed in the component
   */
  tag: PropTypes.object,
  /**
   * Callback function to be called when an item is clicked
   */
  clickItem: PropTypes.func,
};

Tag.defaultProps = {
  tag: null,
  clickItem: null,
};

export default Tag;
