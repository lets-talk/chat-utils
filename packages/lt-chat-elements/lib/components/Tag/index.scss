@import "./config.scss";
.cbox-tag {
  font-size: $font-size-caption;
  @include themify($themes) {
    color: themed('color-primary');
    background-color: themed('color-background-secondary');
  }
  margin: 5px 2px;
  padding: 5px;
  border-radius: 2px;
}
