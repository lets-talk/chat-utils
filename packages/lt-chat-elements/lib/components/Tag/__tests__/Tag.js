import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tag from '../index';

describe('Tag component', () => {
  const tagData = {
    name: 'SPU-CLAVE',
  };

  it('should render with empty props', () => {
    const component = shallow(<Tag />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      data: tagData,
    };

    const component = shallow(<Tag {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
