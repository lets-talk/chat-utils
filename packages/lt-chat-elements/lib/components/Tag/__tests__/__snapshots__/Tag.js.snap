// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component should render with custom props 1`] = `
<StyledTag>
  SPU-CLAVE
</StyledTag>
`;

exports[`Tag component should render with empty props 1`] = `<StyledTag />`;
